docs/src/simulation_tips.md

```diff
@@ -0,0 +1,193 @@
+always work on CPUs, but when their complexity is high (in terms of number of abstract operations)
+the compiler can't translate them into GPU code and they fail for GPU runs. (This limitation is discussed 
+in [this Github issue](https://github.com/CliMA/Oceananigans.jl/issues/1241) and contributors are welcome.)
+For example, in the example below, calculating `u²` works in both CPUs and GPUs, but calculating 
+`KE` only works on CPUs:
+
+```julia
+u, v, w = model.velocities
+u² = ComputedField(u^2)
+KE = ComputedField((u^2 + v^2 + w^2)/2)
+compute!(u²)
+compute!(KE)
+```
+
+There are two approaches to bypass this issue. The first is to nest `ComputedField`s. For example,
+we can make `KE` be successfully computed on GPUs by defining it as
+```julia
+u, v, w = model.velocities
+u² = ComputedField(u^2)
+v² = ComputedField(v^2)
+w² = ComputedField(w^2)
+u²plusv² = ComputedField(u² + v²)
+KE = ComputedField((u²plusv² + w²)/2)
+compute!(KE)
+```
+
+This is a simple workaround that is especially suited for the development stage of a simulation.
+However, when running this, the code will iterate over the whole domain 5 times to calculate `KE`
+(one for each computed field defined), which is not very efficient.
+
+A different way to calculate `KE` is by using `KernelComputedField`s, where the
+user manually specifies the computing kernel to the compiler. The advantage of this method is that
+it's more efficient (the code will only iterate once over the domain in order to calculate `KE`),
+but the disadvantage is that this requires that the has some knowledge of Oceananigans operations
+and how they should be performed on a C-grid. For example calculating `KE` with this approach would
+look like this:
+
+```julia
+using Oceananigans.Operators
+using KernelAbstractions: @index, @kernel
+using Oceananigans.Grids: Center, Face
+using Oceananigans.Fields: KernelComputedField
+
+@inline ψ²(i, j, k, grid, ψ) = @inbounds ψ[i, j, k]^2
+
+@kernel function kinetic_energy_ccc!(tke, grid, u, v, w)
+    i, j, k = @index(Global, NTuple)
+    @inbounds tke[i, j, k] = (
+                              ℑxᶜᵃᵃ(i, j, k, grid, ψ², u) + # Calculates u^2 using function ψ² and then interpolates in x to grid center
+                              ℑyᵃᶜᵃ(i, j, k, grid, ψ², v) + # Calculates v^2 using function ψ² and then interpolates in y to grid center
+                              ℑzᵃᵃᶜ(i, j, k, grid, ψ², w)   # Calculates w^2 using function ψ² and then interpolates in z to grid center
+                             ) / 2
+end
+
+KE = KernelComputedField(Center, Center, Center, kinetic_energy_ccc!, model;
+                         computed_dependencies=(u, v, w))
+```
+
+
+It may be useful to know that there are some kernels already defined for commonly-used diagnostics
+in packages that are companions to Oceananigans. For example
+[Oceanostics.jl](https://github.com/tomchor/Oceanostics.jl/blob/13d2ba5c48d349c5fce292b86785ce600cc19a88/src/TurbulentKineticEnergyTerms.jl#L23-L30)
```

Comment:
I'm a bit hesitant to reference the users to a package that is neither registered nor tested. (mostly the latter tbh)
What do others think?

Reply:
Oceanostics.jl is registered. It could do with some actual tests but plenty of Oceananigans.jl functionality is not tested.

I think it's good to link to packages outside of Oceananigans.jl to give readers the idea that creating their own packages for extra functionality is okay (and actually encouraged!).

I would even advocate for creating an extra page in the docs to link to other packages related to Oceananigans.jl and being liberal with the packages we include.